### PR TITLE
fix(hightlight): do not try to highlight non strings

### PR DESCRIPTION
**Pull request by bas080**

### CODE REVIEW

File: src/component/autocomplete.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class AutoCompleteCustomElement {
 
   //used to determine the string to be shown as option label
   @bindable label = result => {
-    return typeof result === 'object' ? result[this.attribute] : result;
+    return result ? result[this.attribute] : '';
```
Comment (RWOverdijk) [inline src/component/autocomplete.js]:
```js
@bindable label = result => {
  if (typeof result === 'object' && result !== null && result[this.attribute]) {
    return result[this.attribute];
  }

  return '';
}
```

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

File: src/component/autocomplete.js [hunk 2]
```diff
@@ -150,8 +152,8 @@ export class AutoCompleteCustomElement {
   labelWithMatches(result) {
     let label = this.label(result);
 
-    if (!label.replace) {
-      return '';
+    if (typeof label !== 'string') {
+      return ' ';
```
Comment (VMBindraban) [inline src/component/autocomplete.js:156]: Is this space needed? since it should return an empty string.

Comment (bas080) [reply src/component/autocomplete.js:156]: An alternative could be that it returns the string `"no label"`

File: src/component/autocomplete.js [hunk 3]
```diff
@@ -55,7 +55,9 @@ export class AutoCompleteCustomElement {
 
   //used to determine the string to be shown as option label
   @bindable label = result => {
-    return typeof result === 'object' ? result[this.attribute] : result;
+    return (typeof result === 'object' && result !== null && result[this.attribute])
+      ? result[this.attribute]
+      : '';
```
Comment (ppvg) [inline src/component/autocomplete.js]: These elaborate checks on `result` are not necessary if you need to check whether `result[this.attribute]` exists anyway. A simple truthyness check and short circuit should suffice:

```js
return (result && result[this.attribute]) || '';
```

File: src/component/autocomplete.js (hunk 2 above)
Comment (ppvg) [reply src/component/autocomplete.js:156]: So why can't it be an empty string? If it's a workaround, either fix the issue or document why the workaround is needed.

Comment (bas080) [reply src/component/autocomplete.js:156]: I will document that the reason for this is because it avoids the element not getting the proper height.

**Comment by bas080** [comment]
@ppvg, implemented feedback.

**Comment by RWOverdijk** [comment]
@bas080 Is this still needed with all the changes in the other PR? I'm pretty sure I covered these, too. Closing for now.

**Pull request closed without merging**